onto_spread_ed/model/Diff.py

```
@@ -0,0 +1,77 @@
+
+SDiff = Diff[str]
+
+PRIMITIVE_TYPES = (bool, str, int, float, type(None))
+
+
+def diff(a, b, __path="") -> List[Diff]:
+    if not (dataclasses.is_dataclass(a) or isinstance(a, dict)) or not \
+            (dataclasses.is_dataclass(b) or isinstance(b, dict)):
+        raise ValueError(f"Values must be dataclasses or dictionaries. a: '{type(a)}', b: '{type(b)}'")
+
+    if type(a) != type(b):
```

Ruff

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)